server/utils/email/digest.tsx

```diff
@@ -44,6 +45,17 @@ const getAffectedObject = (item: ActivityItem, associations: ActivityAssociation
 	};
 };
 
+const getAffectedObjectUrl = (item: ActivityItem, associations: ActivityAssociations) => {
+	const community = Object.values(associations.community)[0];
+	if (item.pubId) {
+		return pubUrl(community, associations.pub[item.pubId]);
+	}
+	if (item.collectionId) {
+		return collectionUrl(community, associations.collection[item.collectionId]);
+	}
+	return undefined;
```

Comment:
$0.02 — I think we should try to stick to `null` for a falsy fallback value, so `undefined` more strongly carries its built-in meanings of "this key doesn't exist" or "no return value".